projects/aca-folder-rules/src/lib/services/folder-rules.service.ts

```diff
@@ -152,6 +152,16 @@ export class FolderRulesService {
     );
   }
 
+  async getRuleSettings(nodeId: string, key: string = '-isInheritanceEnabled-'): Promise<RuleSettings> {
```

Comment:
Since the rule set service handles the node id, this should be in `FolderRuleSetsService`. We can load it at the same time in the `loadRuleSets` method by adding it in the `combineLatest`, here:
```
switchMap(() => combineLatest(
  this.getMainRuleSet(nodeId),
  loadInheritedRuleSets ? this.getInheritedRuleSets(nodeId) : of([])
))
```
This could become:
```
switchMap(() => combineLatest(
  this.getMainRuleSet(nodeId),
  loadInheritedRuleSets ? this.getInheritedRuleSets(nodeId) : of([]),
  this.getFolderInheritance(nodeId)
))
```
and then using the value in the subscribe to emit to the `isFolderInheritanceEnabledSource` BehaviourSubject.

Reply:
I made `getRuleSettings` more generic, so it could be used in future with other parameters. Do you suggest to redesign it to `getFolderInheritance` for now, so it will only load data for `isInheritanceEnabled$`?

Reply:
Considering it's likely the only rules / rule sets related setting for folders that we have, I don't think it's necessary to make it generic like this. Is this endpoint one of the new private ones? Did you check if there a generic js-api method that could allow you to call this endpoint and then use the folder rule (sets) service to be more specific about `isInheritanceEnabled`?